File: DSA-DAILY/22-07-2025/1695. Maximum Erasure Value@leetcode-med.py
from typing import List

class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        s, currs, res = 0, 0, 0
        seen = set()
        for i in nums:
            while i in seen:
                seen.remove(nums[s])
                currs -= nums[s]
                s += 1
            currs += i
            seen.add(i)
            res = max(res, currs)
        return res

if __name__ == "__main__":
    # Take custom input from user
    nums = list(map(int, input("Enter array elements separated by spaces: ").split()))
    sol = Solution()
    result = sol.maximumUniqueSubarray(nums)
    print("Input array:", nums)
    print("Maximum Erasure Value:", result)

    # Sample Output:
    # Enter array elements separated by spaces: 4 2 4 5 6
    # Input array: [4, 2, 4, 5, 6]
    #

File: DSA-DAILY/DSA-DAILY/22-07-2025/1695. Maximum Erasure Value@leetcode-med.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumUniqueSubarray = function(nums) {
    let seen=new Set();
    let s=0,currs=0,res=0;
    for(let i=0;i<nums.length;i++){
        while(seen.has(nums[i])){
            currs-=nums[s];
            seen.delete(nums[s]);
            s++;
        }
        currs+=nums[i];
        seen.add(nums[i]);
        res=Math.max(res,currs);
    }
    return res;
};

/** 
 * const  nums = [4, 2, 4, 5, 6];
 * const result =maximumUniqueSubarray(nums);
 * console.log("Input array: ",nums);
 * console.log("Maximum Erasable Vaule :",result);
 */

const readline=require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question('Enter the array elements seperated by space: ', input=>{
    const nums=input.trim().split(/\s+/).map(Number);
    const result =maximumUniqueSubarray(nums);
    console.log("Intput array: ",nums);
    console.log("Maximum Ereasable Value: ",result);
    readline.close();
});